src/AMSWorkflow/ams/monitor.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+# Copyright 2021-2023 Lawrence Livermore National Security, LLC and other
+# AMSLib Project Developers
+#
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import datetime
+import json
+import logging
+import multiprocessing
+import threading
+import time
+from typing import Callable, List, Union
+
+
+class AMSMonitor:
+    """
+    AMSMonitor can be used to decorate class methods and will
+    record automatically the duration of the tasks in a hashmap
+    with timestamp. The decorator will also automatically
+    record the values of all attributes of the class.
+
+        class ExampleTask1(Task):
+            def __init__(self):
+                self.total_bytes = 0
+                self.total_bytes2 = 0
+
+            # Example: we do not want to record total_bytes
+            # but just total_bytes2
+            #
+            # @AMSMonitor() would record all attributes
+            # (total_bytes and total_bytes2)
+            #
+            # @AMSMonitor(accumulate=True) would record all
+            #   attributes and accumulate their values
+            #   (sum up total_bytes instead of recording a
+            #   different total_bytes per invocation)
```

Comment:
Does this mean that by default monitor captures all attributes, and when you specify something it only captures the specified values? Can you make the comment clear?

Reply:
I updated the comments, thanks.